File: backend/src/modules/activities/domain/submission.ts
export type SubmissionStatus = 'PENDING' | 'SUBMITTED' | 'GRADED';

export type Submission = {
  id: string;
  teacherId: string;
  activityId: string;
  studentId: string;
  groupId: string | null;
  status: SubmissionStatus;
  score: number | null;
  observations: string | null;
  submittedAt: Date | null;
  gradedAt: Date | null;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date | null;
};

File: frontend/lib/domain/entities/submission.dart
class Submission {
  const Submission({
    required this.id,
    required this.activityId,
    required this.studentId,
    this.groupId,
    required this.status,
    this.score,
    this.observations,
    this.studentName,
  });

  final String id;
  final String activityId;
  final String studentId;
  final String? groupId;
  final String status;
  final double? score;
  final String? observations;

  /// Preenchido pela tela ao cruzar com a lista de matrículas da turma.
  final String? studentName;

  bool get isGraded => status == 'GRADED';

  Submission copyWithStudentName(String? name) => Submission(
        id: id,
        activityId: activityId,
        studentId: studentId,
        groupId: groupId,
        status: status,
        score: score,
        observations: observations,
        studentName: name ?? studentName,
      );
}
